app/interactors/build_user_with_membership.rb

```diff
@@ -1,6 +1,6 @@
 class BuildUserWithMembership
```

Comment:
May be a touch pendantic but - could we maybe think of a better name for this class? Feels like it should be named something more object-y rather than an action.